python/ray/air/checkpoint.py

```diff
@@ -199,6 +219,16 @@ def __repr__(self):
         parameter, argument = self.get_internal_representation()
         return f"{self.__class__.__name__}({parameter}={argument})"
 
+    @property
+    @DeveloperAPI
+    def metadata(self) -> CheckpointMetadata:
+        return CheckpointMetadata(
+            checkpoint_type=self.__class__,
+            checkpoint_state={
+                attr: getattr(self, attr) for attr in self._SERIALIZED_ATTRS
+            },
```

Comment:
Need ray version here?

Reply:
The Ray version is set when the `CheckpointMetadata` class is defined, so there's no need to explicitly pass a version to the `ray_version` parameter. Although, it might be easier to understand if we explicitly set the version during construction. What do you think?